docs/api/effector/sample.md

```diff
@@ -256,3 +256,62 @@ clickButton('click B')
 ```
 
 [try it](https://share.effector.dev/yI70z0nd)
+
+# `sample(sourceStore)`
+
+Shorthand for `sample({ source: sourceStore, clock: sourceStore })`, it can be used to make updates of `sourceStore` non-greedy, thus batching updates of `sourceStore`.
+
+This is especially useful if we are combining different stores, and resulting store switches its state multiple times within single update. `sample` ensures it will fire only upon the last state
+
+#### Arguments
+
+1. `sourceStore` ([_Store_](Store.md)): Source store
+
+#### Returns
+
+([_`Store`_](Store.md)) - Non-greedy store
+
+#### Example 1
+
+```js try
+import {createStore, createEffect, sample, combine} from 'effector'
+
+const data = [{name: 'physics', id: 1}]
+
+const fetchContent = createEffect({
+  handler: () => new Promise(resolve => setTimeout(() => resolve(data), 0)),
+})
+
+const $lessonIndex = createStore(0)
+const $allLessons = createStore([]).on(
```

Comment:
Tip: you can use `restore` here
https://share.effector.dev/bdAohSKM

https://github.com/zerobias/effector/blob/master/src/effector/restore.js#L34

Reply:
Sweet, didn't know about that, should I open PR for that? This one got already merged

Reply:
Or I guess, you can just push it to master, simple fix

Reply:
Just a note for the future